Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -209,21 +227,91 @@ namespace AzToolsFramework
                     if (AZ::Geometry2DUtils::ShortestDistanceSqPointSegment(AZ::Vector2(points[index]), previousCenter2D, center2D) <=
                          manipulatorRadiusSq)
                     {
-                        intensities[index] = intensity;
-                        opacities[index] = opacity;
-                        validFlags[index] = true;
-                    }
-                    else
-                    {
-                        intensities[index] = 0.0f;
-                        opacities[index] = 0.0f;
-                        validFlags[index] = false;
+                        validPoints.emplace_back(points[index]);
+                        intensities.emplace_back(intensity);
+                        opacities.emplace_back(opacity);
                     }
                 }
             });
 
+            // Set the blending operation based on the current paintbrush blend mode setting.
+            PaintBrushNotifications::BlendFn blendFn;
+            switch (blendMode)
+            {
+                case PaintBrushBlendMode::Normal:
```

Comment:
This is one of the few times I'm deliberately _not_ making a more extensible approach. :) The design behind the paint brush tool is that it contains all of the paint brush logic possible. Other systems shouldn't need to extend with custom blend types. There's a finite set of blend modes defined in paint programs.

That said, once we have a second tangible use case for the paint brush, it's possible that we'll find a reason to add custom types, and it could extracted at that point. It just feels like overkill to add that support right now, when the intent is _not_ to have a need for it.

As for the blend functions being the same as mixed gradient, I'm definitely open to suggestions here. I stared at it a bunch but didn't see a great way to share them. The blend math itself is the same, but the lerping/clamping is a bit different between the two use cases. I just wasn't seeing enough logic to pull out into shared code, especially if we also keep the enums separate between the two.